modules/photo/src/merge.cpp

```diff
@@ -342,7 +342,7 @@ class MergeRobertsonImpl CV_FINAL : public MergeRobertson
             LUT(images[i], response, im);
 
             result += times.at<float>((int)i) * w.mul(im);
-            wsum += times.at<float>((int)i) * times.at<float>((int)i) * w;
+            wsum += times.at<float>((int)i) * times.at<float>((int)i) * (w + DBL_EPSILON);
         }
         result = result.mul(1 / wsum);
```

Comment:
Patching `wsum` doesn't look valid.

Patch divide operation instead (`Scalar::all()` is necessary as we work with multi-channel data):
```
result = result.mul(1 / (wsum + Scalar::all(DBL_EPSILON)));
```

Reply:
@alalek Ah good catch. It technically works, but a warning pops up:

```
assign OpenCV/MatExpr: processing of multi-channel arrays might be changed in the future: https://github.com/opencv/opencv/issues/16739
```

Addressed the issue and actually moved the epsilon to where the division occurs. Thank you!